libvmaf/src/feature/luminance_tools.c

```diff
@@ -65,13 +66,38 @@ int vmaf_luminance_init_luma_range(VmafLumaRange *luma_range, int bitdepth, enum
     return err;
 }
 
+int vmaf_luminance_init_eotf(VmafEOTF *eotf, char* eotf_str) {
+    if (strcmp(eotf_str, "bt1886") == 0) {
+        *eotf = vmaf_luminance_bt1886_eotf;
+    }
+    else if (strcmp(eotf_str, "pq") == 0) {
+        *eotf = vmaf_luminance_pq_eotf;
+    }
+    else {
+        vmaf_log(VMAF_LOG_LEVEL_ERROR, "unknown EOTF received");
+        return -EINVAL;
+    }
+    return 0;
+}
+
 double vmaf_luminance_bt1886_eotf(double V) {
     double a = pow(pow(BT1886_LW, 1.0 / BT1886_GAMMA) - pow(BT1886_LB, 1.0 / BT1886_GAMMA), BT1886_GAMMA);
     double b = pow(BT1886_LB, 1.0 / BT1886_GAMMA) / (pow(BT1886_LW, 1.0 / BT1886_GAMMA) - pow(BT1886_LB, 1.0 / BT1886_GAMMA));
     double L = a * pow(MAX(V + b, 0), BT1886_GAMMA);
     return L;
 }
 
+double vmaf_luminance_pq_eotf(double V) {
+    double m_1 = 0.1593017578125;
+    double m_2 = 78.84375;
+    double c_1 = 0.8359375;
+    double c_2 = 18.8515625;
+    double c_3 = 18.6875;  // c_3 = c_1 + c_2 - 1
+    double num = MAX(0, pow(V, 1.0 / m_2) - c_1);
+    double den = c_2 - c_3 * pow(V, 1.0 / m_2);
```

Comment:
These can probably be declared as const, and for `num`, it might be better to separate out the `pow` call from the `MAX` macro, so it's not called multiple times

Reply:
Thanks for your comments, the last commit addresses them.